presto-main/src/test/java/com/facebook/presto/type/TestTimestampWithTimeZone.java

```diff
@@ -113,4 +133,41 @@ public void testCastToTimestamp()
                 TIMESTAMP,
                 sqlTimestampOf(2001, 1, 22, 3, 4, 5, 321, session));
     }
+
+    @Test
+    public void testCastInvalidTimestamp()
+    {
+        assertInvalidFunction("CAST('ABC' AS TIMESTAMP WITH TIME ZONE)", INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp with time zone: ABC");
+        assertInvalidFunction("CAST('2022-01-00 00:00:00 UTC' AS TIMESTAMP WITH TIME ZONE)", INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp with time zone: 2022-01-00 00:00:00 UTC");
```

Comment:
The UTC test cases here are confusing. The reason this and the following one fail is not related to the timezone since UTC is a valid timezone. The reason is the invalid date. I guess you can put `Invalid date` as part of the description.